skt/reporter.py

```diff
@@ -234,6 +234,24 @@ def __init__(self, cfg):
         # multireporting
         self.multi_job_ids = []
 
+    def get_config_local(self):
+        """Retrieve a local config file from the disk."""
+        config_file = "{}/.config".format('workdir')
+        if os.path.isfile(config_file):
+            # The config file
+            with open(config_file, "r") as fileh:
+                return fileh.read()
+
+        return None
+
+    def get_config_remote(self):
+        """Retrieve a config file from a remote location."""
+        response = requests.get(self.cfg.get("cfgurl"))
+        if response.status_code == requests.codes.ok:
+            return response.text
```

Comment:
Hmm, shouldn't we throw an exception if we get anything but 200 OK here?

Reply:
A 301/302 redirect would be fine and requests should follow those. For example, the published location might redirect 80 to 443 and I doubt we should throw an exception for that.

Reply:
If redirecting is allowed, the `status_code` is the status of the last redirect, which should be `OK`, not 3XX.

Reply:
Well, if it's a 30x code, then we shouldn't be returning `response.text` in any case. Instead we need to follow the redirect, and raise an exception if we couldn't get the configuration.